miasm2/ir/ir.py

```diff
@@ -278,25 +300,34 @@ def _get_dst(self):
         self._dst_linenb = final_linenb
         return final_dst
 
-    def _set_dst(self, value):
-        """Find and replace the IRDst affectation's source by @value"""
-        if self._dst_linenb is None:
-            self._get_dst()
+    @property
+    def dst(self):
+        """Find the IRDst affectation and update dst, dst_linenb accordingly"""
+        if self.is_dst_set():
+            return self._dst
+        return self.cache_dst()
 
-        new_assignblk = dict(self.irs[self._dst_linenb])
-        for dst in new_assignblk:
-            if dst.is_id("IRDst"):
-                new_assignblk[dst] = value
-                # Sanity check is already done in _get_dst
-                break
-        self._dst = value
-        instr = self.irs[self._dst_linenb].instr
-        self.irs[self._dst_linenb] = AssignBlock(new_assignblk, instr)
-    dst = property(_get_dst, _set_dst)
+    def set_dst(self, value):
+        """Generate a new IRBlock with a dst fixed to @value"""
```

Comment:
May a warning (in the docstring) saying the previous value of `IRDst`, if any, will be overwritten.